File: 0035. Search Insert Position/main0035.cpp

#include <iostream>
#include <string>
#include<vector>
using namespace std;


class main0035 {
  public:
  int searchInsert(vector<int>& nums, int target) {
    int pivot, left = 0, right = nums.size() - 1;
    while (left <= right) {
      pivot = left + (right - left) / 2;
      if (nums[pivot] == target) return pivot;
      if (target < nums[pivot]) right = pivot - 1;
      else left = pivot + 1;
    }
    return left;
  }
};

int main() {
    main0035 sol;
    int n[] = {1,3,5,6};
    int target = {5};
    vector<int> nums(n,n+4);
    int res = sol.searchInsert(nums, target);
    cout << res;
    return 0;
};

File: 0035. Search Insert Position/main0035.java
import java.util.Arrays;

public class main0035 {
    public static void main(String[] args) {
        Solution0035 sol = new Solution0035();
        int nums[], target, res;

        nums = new int[]{1,3,5,6};
        target = 5;
        res = sol.searchInsert(nums, target);
        System.out.println(res);
    }
}

class Solution0035 {
    public int searchInsert(int[] nums, int target) {
        int index = Arrays.binarySearch(nums, target);
        if (index>=0) {
            return index;
        }
        else {
            return (-(index) - 1);
        }
    }
}